basic_datatypes.ipynb

```diff
@@ -0,0 +1,2329 @@
+    "\n",
+    "def solution_return_first_and_last_element_from_list(my_list):\n",
+    "    # Your code starts here\n",
+    "    return\n",
+    "    # Your code ends here"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%ipytest\n",
+    "\n",
+    "def solution_first_and_last_element_are_equal(my_list):\n",
+    "    # Your code starts here\n",
+    "    return\n",
+    "    # Your code ends here"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%ipytest\n",
+    "\n",
+    "def solution_lists_are_equal(list1, list2):\n",
+    "    # Your code starts here\n",
+    "    return\n",
+    "    # Your code ends here"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%ipytest\n",
+    "\n",
+    "def solution_lists_are_equal_but_not_same(list1, list2):\n",
+    "    # Your code starts here\n",
+    "    return\n",
+    "    # Your code ends here"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%ipytest\n",
+    "\n",
+    "def solution_greater_or_equal(list1, list2):\n",
+    "    # Your code starts here\n",
+    "    return\n",
+    "    # Your code ends here"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Tuple\n",
+    "\n",
+    "Tuple is similar to a list in many ways.\n",
+    "However, the tuple is immutable, which means that it cannot be changed after it is created.\n",
+    "The tuple is defined by round brackets `()` and its elements are separated by commas `,`.\n",
+    "The elements of the tuple can be accessed by their index in the same way as the list.\n",
+    "The advantage of the tuple over the list is that it is faster and more memory efficient, which makes it a good choice for storing data that does not change.\n",
+    "The disadvantage of the tuple is that it cannot be changed after it is created, and it does not have many methods.\n",
+    "\n",
+    "Let's see how the tuple works in practice:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "my_tuple = (1, 2.5, True, None)\n",
+    "\n",
+    "print(f\"my_tuple is {my_tuple}\")\n",
+    "print(f\"The first element of 'my_tuple' is {my_tuple[0]} and is of type {type(my_tuple[0])}\")\n",
+    "print(f\"The last element of 'my_tuple' is {my_tuple[-1]} and is of type {type(my_tuple[-1])}\")\n",
+    "print(f\"The length of 'my_tuple' is {len(my_tuple)}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Unlike lists, sets and dictionaries, tuples do not require parentheses.\n",
```

Comment:
For completeness, just add a line saying that parentheses are required with tuples only when there's an ambiguity. You can say that we will discuss an example in "Functions".

Reply:
done in https://github.com/empa-scientific-it/python-tutorial/pull/14/commits/6ef5c319587543e6b409971dcb4e837b8e623085